src/common/message-creators/action-message-dispatcher.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { PayloadWithEventName } from '../../background/actions/action-payloads';
+import { Message } from '../message';
+import { Messages } from '../messages';
+import { TelemetryData } from '../telemetry-events';
+
+export class ActionMessageDispatcher {
+    constructor(private postMessageDelegate: (message: Message) => void, private tabId: number) {}
+
+    public dispatchMessage(message: Message): void {
```

Comment:
The reuse of the same `Message` type feels misleading, here; it isn't apparent to me as a reader from the names/types here that the caller of `dispatchMessage` should not fill in the `tabId` property of the input object.

Reply:
I added a related type: `InterpreterMessage` that "extend" `Message` and add an optional `tabId` property. Now `Message` doesn't have the `tabId?` property. In a following PR I'll rename `Message` to `ActionMessage` to match the naming of our `XActionMessageCreator`'s

Reply:
how about having two message types? GlobalMessage & TabContextMessage?

---
In reply to: [274691083](https://github.com/Microsoft/accessibility-insights-web/pull/517#discussion_r274691083) [](ancestors = 274691083)